src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -0,0 +1,338 @@
+        return result;
+    }
+
+    /**
+     * Performs substitutions defined for this transform, and returns the result.
+     */
+    public String performSubstitutions(String inputString) {
+        if (substitutions != null && substitutions.size() != 0) {
+            String result = inputString;
+            for (Substitution substitution : substitutions) {
+                result = substitution.replace(result);
+            }
+            return result;
+        }
+        return inputString;
+    }
+
+    public enum NormalizeOperation {
+        SENTENCE_CASE, TITLE_CASE
+    }
+
+    /**
+     * This class holds the regex/replacement pair, and the cached compiled regex pattern.
+     */
+    public static class Substitution implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        /** The regex pattern string to match. */
+        public String pattern;
+        /** The string that should replace regex (see effectiveReplacement below). */
+        public String replacement;
+        /** true if whitespace surrounding regex should be create or normalized to one space. */
+        public boolean normalizeSpace;
+
+        private Pattern patternObject;
+        /**
+         * Replacement string that is actually used for the substitution,
+         * and set according to the value of normalizeSpace.
+         */
+        private String effectiveReplacement;
+
+        /** Empty constructor needed for persistence */
+        public Substitution() {}
+
+        public Substitution(String pattern, String replacement) {
+            this(pattern, replacement, false);
+        }
+
+        public Substitution(String pattern, String replacement, boolean normalizeSpace) {
+            this.pattern = pattern;
+            this.normalizeSpace = normalizeSpace;
+            this.replacement = replacement;
+        }
+
+        /**
+         * Pre-compiles the regex pattern and determines the actual replacement string
+         * according to normalizeSpace.
+         */
+        private void initialize() {
+            if (normalizeSpace) {
+                // If normalizeSpace is set, reduce spaces before and after the regex to one space,
+                // or insert one space before and one space after if there is none.
+                // Note: if the replacement must be blank, then normalizeSpace should be set to false
+                // and whitespace management should be handled in the regex instead.
+                this.patternObject = Pattern.compile(String.format("\\s*%s\\s*", pattern));
+                this.effectiveReplacement = String.format(" %s ", replacement);
+            } else {
+                this.patternObject = Pattern.compile(pattern);
+                this.effectiveReplacement = replacement;
+            }
+        }
+
+        /**
+         * Perform the replacement of regex in the provided string, and return the result.
+         */
+        public String replace(String input) {
```

Comment:
Nit: Should this be named replaceAll to match the method it invokes? Not 💯 on this... just a thought.

Reply:
Updated in https://github.com/ibi-group/datatools-server/pull/371/commits/b2dd61e73bfedc146412d3c3c9386693d5799609